### ISSUE TITLE
Change message displayed for profiles on limited instances

### ISSUE THREAD

**Issue by leahoswald** [mastodon/mastodon#28436]
### Steps to reproduce the problem

Option 1:
1. Limit an instance
2. Browse a profile on this limited instance you are not following
3. See the message "This profile has been hidden by the moderators of example.com."

Option 2:
1. Limit an instance
2. Get a follow request from an account on this instance even if your account is public
3. Click on the profile 
4. See the message "This profile has been hidden by the moderators of example.com."

### Expected behaviour

A message that make it clear to the user what happend

### Actual behaviour

A message that leads to a lot of questions by the users

### Detailed description

If you limit an instance, for example due to a spam incident, users get a message like "This profile has been hidden by the moderators of example.com." which raises a lot of questions by users if they see it while browsing profiles or getting a follow request for their public account. If would be great to change the text in something helpful explaining what happened. Minimal example example: "The account or the server hosting this account was [limited](link to docs explaining what limiting means) by the admins of example.com." or more detailed "The account or the server hosting this account was [limited](link to docs explaining what limiting means) by the admins of example.com. Therefore you see the posts of this account only if you follow the specific account and you have to a approve follow request even if your account is public. If you want to see it just now click 'Show profile anyway'"

### Mastodon instance

chaos.social

### Mastodon version

v4.2.3

### Browser name and version

Firefox

### Operating system

Linux

**Comment by ryliejamesthomas** [comment]
Have definitely had users confused by this message.

Perhaps it also could display the text from the 'Public reason' field, if there is one.

**Comment by masukomi** [comment]
building on this with another concrete example. 
One of the servers I'm on has limited mastodon.social (reasons not relevant). However, when I view a profile from there it says 

> This profile has been hidden by the moderators of <server name here>

There are 2 problems with this. 

1. it's a lie. Our moderators didn't hide the profile. They limited the server, and socially this is a VERY significant difference.  It's the difference between "we've closed our borders to folks from country X" versus "this is a dangerous individual"
2. it misleads users into thinking that the user / profile in question has done something "bad", & that the leaders of their community (their server) have decided that this person doesn't adhere to community values.

It's just not true. In our case, we noted that there are a lot of problematic actors on mastodon.social, BUT we also noted that it's not an inherently bigoted server we should defederate from. There are a lot of perfectly decent people there, who aren't bigots. 

The warning shouldn't make innocent people appear as if they've violated your communities values when they haven't done anything "bad".

**Comment by ShadowJonathan** [comment]
Bump, as we (tech.lgbt) have been getting users to ask why a certain person's profile has been limited, when we have limited the entire instance. Being able to show a differing message based on whether an instance was limited, or a user, would be very useful.

**Comment by Laquinh** [comment]
I agree this would be a good idea, this message showed up when looking at my profile from another instance and I got scared.

I like the first minimal example much better than the detailed one.

**Comment by yaomtc** [comment]
Also confused why this comic dude's profile displayed as "hidden"? This is viewing it from an account on the same instance. https://mastodon.social/@damnyouwillis
